### PR TITLE
Update NFT Metadata + Remove /node_modules

### PR DESCRIPTION
**Pull request by alliu930410**

### CODE REVIEW

**Comment by alliu930410** [comment]
Comment from the Flow Team:

https://github.com/KLKTN/nfts/commit/2921c86a10f96c523b4ec014070b248e3aaba5ec

94: `    pub let metadata: {String: String}`

This will be publicly settable by the current owner of the NFT.

If you do not wish this, make it access(self) and provide a function
that returns a copy of it, e.g.
```
pub fun getMetadata(): {String: String} {
    return self.metadata
}
```

But see 100, below.

100: `     self.metadata = KlktnNFT.klktnNFTTypeSet[initTypeID]!.metadata`

This copies the dictionary rather than referencing it, so unless the
metadata will be mutated on a per-NFT level
I would remove this, and the metadata field from the NFT, then instead
provide either a contract-level or
NFT-level accessor that fetches the type data for a given NFT id if it exists.

See also 94 above.

**Comment by alliu930410** [comment]
@flockonus I've updated the NFT resource per Flow team's suggestion: we will not have the metadata field in NFTs themselves but rather a getter function to retrieve a copy of the metadata of the NFT. Let me know your thoughts on this (account space saved is around 100 per mint from 250 ---> 150)
Also added a unit test trying to mutate the metadata that would not succeed (https://github.com/KLKTN/nfts/pull/2/commits/e1e2ef2107f0ef86e007e9e72ffbd477b9509788).

File: cadence/transactions/try_mutate_metadata.cdc [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    .borrow<&KlktnNFT.Collection{KlktnNFT.KlktnNFTCollectionPublic}>()
+    ?? panic("Could not borrow capability from public collection")
+
+    // Borrow a reference to a specific NFT in the Collection
+    let klktnNFT = collectionRef.borrowKlktnNFT(id: tokenID)
+    ?? panic("No such token in that collection")
+
+    let metadata = klktnNFT.getNFTMetadata()
+    metadata["extra"] = "345"
+    metadata["releaseYear"] = "2039"
```
Comment (flockonus) [inline cadence/transactions/try_mutate_metadata.cdc:22]: It took me a bit to understand why this tx doesn't throw, but we're editing properties on a copy of the data eh.
So flow can return a `{string:string}` from a function that is a copy. It's coherent with the concept of resource safety I guess.

**Comment by flockonus** [comment]
@alliu930410 impressive storage savings 👍 
I believe in the future we'll need to change data on particular NFTs, for instance in order to stamp a particular moment with a particular photo or something like that, but for now this is great as it saves storage for every mint!

**Review by flockonus — APPROVED** [review]
LGTM!

**Pull request merged**